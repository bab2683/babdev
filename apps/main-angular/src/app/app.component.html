<!-- <babdev-sidebar #sidebar (currentStatus)="sidebarStatus($event)" [openedAtInit]="true">
  <ng-container sidebarContent>
    sidebar content here
  </ng-container>

  <div class="page" pageContent>
    <header>
      header
    </header>
    <main class="page__content">
      main content here
      <router-outlet></router-outlet>
    </main>
    <footer>
      footer
    </footer>
  </div>
</babdev-sidebar> -->

<babdev-sidebar #sidebar [openedAtInit]="true" [withTouch]="true">
  <ng-container sidebarContent>
    <babdev-page-list></babdev-page-list>
  </ng-container>

  <div class="page" pageContent>
    <main class="page__content">
      <div class="page__router" [@routerAnimations]="prepareRoute(outlet)">
        <router-outlet #outlet="outlet"></router-outlet>
        <!-- main content here common library test
        <br />
        <span>{{ 'common.TIME.DAY.SINGLE' | translate | async }}</span>
        <br />
        <button (click)="translateService.changeActiveLanguage('en')">Change to en</button
        ><br /><button (click)="translateService.changeActiveLanguage('it')">Change to it</button> -->
      </div>
    </main>
    <footer>
      footer
    </footer>
  </div>
</babdev-sidebar>
