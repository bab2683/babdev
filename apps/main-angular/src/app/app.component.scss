.page {
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  &__content {
    display: flex;
    flex-grow: 1;
    position: relative;
  }
}
