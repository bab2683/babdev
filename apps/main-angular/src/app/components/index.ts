export * from './header/header.component';
export * from './page-list/page-list.component';
