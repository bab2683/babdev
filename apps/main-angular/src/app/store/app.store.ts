import { routerReducer, RouterReducerState } from '@ngrx/router-store';
import { ActionReducerMap, MetaReducer } from '@ngrx/store';

import { environment } from '@env/environment';
import {
  GlobalEffects,
  globalReducer,
  GlobalState,
  GLOBAL_FEATURE,
  ROUTER_FEATURE
} from './features';

export interface AppState {
  [GLOBAL_FEATURE]: GlobalState;
  [ROUTER_FEATURE]: RouterReducerState;
}

export const AppEffects: any[] = [GlobalEffects];

export const reducers: ActionReducerMap<AppState> = {
  [GLOBAL_FEATURE]: globalReducer,
  [ROUTER_FEATURE]: routerReducer
};

export const metaReducers: Array<MetaReducer<
  AppState
>> = !environment.production ? [] : [];
