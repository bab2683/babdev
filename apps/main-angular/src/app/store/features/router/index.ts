export * from './router.selectors';
export * from './router.state';
