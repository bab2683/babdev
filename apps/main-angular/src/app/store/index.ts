export * from './app.store';
export * from './features';
export * from './mocks';
