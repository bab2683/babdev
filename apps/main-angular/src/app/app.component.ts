import { isPlatformBrowser } from '@angular/common';
import { Component, Inject, PLATFORM_ID, ViewChild } from '@angular/core';
import { RouterOutlet } from '@angular/router';
import { SidebarComponent, SidebarStatus } from '@babdev/sidebar';
import { TranslateService } from '@babdev/translate';
import { isMobile } from '@babdev/utils';

import { fade } from './pages/route.animations';

@Component({
  selector: 'babdev-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  animations: [fade]
})
export class AppComponent {
  @ViewChild('sidebar', { static: true }) sidebar: SidebarComponent;

  title = 'main-angular';
  public isBrowser: boolean;
  public isMobile: boolean;

  constructor(
    public translateService: TranslateService,
    @Inject(PLATFORM_ID) private platformId: string
  ) {
    this.isBrowser = isPlatformBrowser(this.platformId);
    this.isMobile = isMobile(this.isBrowser);

    if (this.isBrowser && !this.isMobile) {
      document.body.classList.add('isDesktop');
    }
  }

  public prepareRoute(outlet: RouterOutlet) {
    return outlet && outlet.activatedRouteData && outlet.activatedRouteData['name'];
  }

  // sidebarStatus(status: SidebarStatus) {
  //   console.log('sidebar status', status);
  // }
}
