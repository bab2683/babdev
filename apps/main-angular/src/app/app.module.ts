import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule } from '@angular/router';
import { RequestModule } from '@babdev/request';
import { SidebarModule } from '@babdev/sidebar';
import { TranslateModule } from '@babdev/translate';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';

import { translateConfig } from '@constants';

import { AppComponent } from './app.component';
import { PageListComponent } from './components/page-list/page-list.component';

@NgModule({
  declarations: [AppComponent, PageListComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    FontAwesomeModule,
    RequestModule,
    RouterModule.forRoot([], { initialNavigation: 'enabled' }),
    SidebarModule,
    TranslateModule.forRoot(translateConfig)
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule {}
