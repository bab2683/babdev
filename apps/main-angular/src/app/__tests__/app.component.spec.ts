import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
import { Renderer2, Type } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateService } from '@babdev/translate';
import { TranslatePipeMock, TranslateServiceMock } from '@babdev/translate-testing';
import { Store } from '@ngrx/store';
import { MockStore, provideMockStore } from '@ngrx/store/testing';

import { AppState, MockedAppStore } from '@store';

import { AppComponent } from '../app.component';

class MockRenderer {
  addClass: jest.Mock = jest.fn();
}

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let store: MockStore<AppState>;
  const initialState: AppState = MockedAppStore;
  let renderer: Renderer2;

  // beforeEach(() => {
  //   renderer = new MockRenderer();
  // });

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, NoopAnimationsModule],
      declarations: [AppComponent, TranslatePipeMock],
      providers: [
        {
          provide: TranslateService,
          useClass: TranslateServiceMock
        },
        provideMockStore({ initialState }),
        Renderer2
      ],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.debugElement.componentInstance;
    store = TestBed.get<Store<AppState>>(Store);
    renderer = fixture.componentRef.injector.get<Renderer2>(Renderer2 as Type<Renderer2>);

    jest.spyOn(renderer, 'addClass');
  });

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it(`should have as title 'main-angular'`, () => {
    expect(component.title).toEqual('main-angular');
  });

  it('should assing the global class', () => {
    store.setState({ global: { isMobile: false } });
    component.ngOnInit();

    expect(component.isMobile).toEqual(false);
    expect(renderer.addClass).toHaveBeenCalled();
  });

  it('should not assing the global class', () => {
    store.setState({ global: { isMobile: true } });
    component.ngOnInit();

    expect(component.isMobile).toEqual(true);
    expect(renderer.addClass).not.toHaveBeenCalled();
  });
});
