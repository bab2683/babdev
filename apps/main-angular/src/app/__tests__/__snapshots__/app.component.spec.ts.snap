// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the snapshot 1`] = `
<babdev-root
  isBrowser={[Function Boolean]}
  isMobile="false"
  platformId={[Function String]}
  sidebar={[Function ElementRef]}
  title={[Function String]}
  translateService={[Function TranslateServiceMock]}
>
  <babdev-sidebar
    class="ng-tns-c1-0"
  >
    
    <babdev-page-list
      class="ng-tns-c1-0"
    />
    <div
      class="page"
      pagecontent=""
    >
      <header
        class="ng-tns-c1-0"
      >
         header 
      </header>
      <main
        class="page__content"
      >
        <div
          class="page__router ng-trigger ng-trigger-routerAnimations ng-animating"
          style=""
        >
          <router-outlet
            class="ng-tns-c1-0"
          />
        </div>
      </main>
      <footer
        class="ng-tns-c1-0"
      >
         footer 
      </footer>
    </div>
  </babdev-sidebar>
</babdev-root>
`;
